src/main/java/org/folio/service/protection/AcquisitionsUnitsServiceImpl.java

```diff
@@ -28,92 +29,97 @@ public class AcquisitionsUnitsServiceImpl extends BaseService implements Acquisi
   protected final Logger logger = LogManager.getLogger(this.getClass());
 
   @Override
-  public CompletableFuture<AcquisitionsUnitCollection> getAcquisitionsUnits(String query, int offset, int limit, String lang, Context context, Map<String, String> headers) {
+  public Future<AcquisitionsUnitCollection> getAcquisitionsUnits(String query, int offset, int limit, String lang, Context context, Map<String, String> headers) {
     logger.debug("getAcquisitionsUnits:: Trying to get acquisition units with query: {}, offset: {}, limit: {}", query, offset, limit);
-    HttpClientInterface client = getHttpClient(headers);
+    RequestContext requestContext = new RequestContext(context, headers);
+    Promise<AcquisitionsUnitCollection> promise = Promise.promise();
     if (StringUtils.isEmpty(query)) {
       query = ACTIVE_UNITS_CQL;
     } else if (!query.contains(IS_DELETED_PROP)) {
       query = combineCqlExpressions("and", ACTIVE_UNITS_CQL, query);
     }
     String endpoint = String.format(GET_UNITS_BY_QUERY, limit, offset, buildQuery(query, logger), lang);
-    return handleGetRequest(endpoint, client, headers, logger)
-      .thenApply(jsonUnits -> jsonUnits.mapTo(AcquisitionsUnitCollection.class))
-      .handle((acqUnitsColl, t) -> {
-        client.closeClient();
+    return handleGetRequest(endpoint, requestContext, logger)
+      .compose(jsonUnits -> {
+        AcquisitionsUnitCollection acquisitionsUnitCollection = jsonUnits.mapTo(AcquisitionsUnitCollection.class);
+        promise.complete(acquisitionsUnitCollection);
```

Comment:
No need to use separate promise variable. You could return the response directly using `.map(...)` method

Reply:
fixed